app/controllers/api/v1/projects_controller.rb

```diff
@@ -61,13 +62,17 @@ def activities
       def project_params
         raise TypeError unless params.require(:data).require(:type) == 'projects'
 
-        params.require(:data).require(:attributes).permit(:name, :visibility, :archived)
+        params.from_jsonapi.require(:project).permit(:name, :visibility, :archived, :project_folder_id)
       end
 
       def load_project_for_managing
         @project = @team.projects.find(params.require(:id))
         raise PermissionError.new(Project, :manage) unless can_manage_project?(@project)
       end
+
+      def check_parent_folder
+        @team.project_folders.find(project_params[:project_folder_id]) if project_params[:project_folder_id]
```

Comment:
Shouldn't it be done by model validator?

Reply:
I guess yes